R/admiral_options.R

```diff
@@ -0,0 +1,166 @@
+#'   `subject_keys` argument in all admiral functions.
+#'
+#' @param set_default If set to `TRUE` will restore admiral defaults.
+#'
+#' @details
+#' Modify an admiral option, e.g `subject_keys`, such that it automatically affects downstream
+#' function inputs where `get_admiral_option()` is called such as `derive_param_exist_flag()`.
+#'
+#' @author Zelos Zhu
+#'
+#' @return
+#' No return value, called for side effects.
+#'
+#' @keywords admiral_options
+#' @family admiral_options
+#'
+#' @export
+#'
+#' @seealso [vars()], [get_admiral_option()], [derive_param_exist_flag()],
+#' [derive_param_first_event()], [derive_param_tte()], [derive_var_disposition_status()],
+#' [derive_var_dthcaus()], [derive_var_extreme_dtm()], [derive_vars_disposition_reason()],
+#' [derive_vars_period()], [create_period_dataset()]
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#' library(lubridate)
+#' library(tibble)
+#' set_admiral_options(subject_keys = vars(STUDYID, USUBJID2))
+#'
+#' # Derive a new parameter for measurable disease at baseline
+#' adsl <- tribble(
+#'   ~USUBJID2,
+#'   "1",
+#'   "2",
+#'   "3"
+#' ) %>%
+#'   mutate(STUDYID = "XX1234")
+#'
+#' tu <- tribble(
+#'   ~USUBJID2, ~VISIT,      ~TUSTRESC,
+#'   "1",       "SCREENING", "TARGET",
+#'   "1",       "WEEK 1",    "TARGET",
+#'   "1",       "WEEK 5",    "TARGET",
+#'   "1",       "WEEK 9",    "NON-TARGET",
+#'   "2",       "SCREENING", "NON-TARGET",
+#'   "2",       "SCREENING", "NON-TARGET"
+#' ) %>%
+#'   mutate(
+#'     STUDYID = "XX1234",
+#'     TUTESTCD = "TUMIDENT"
+#'   )
+#'
+#' derive_param_exist_flag(
+#'   dataset_adsl = adsl,
+#'   dataset_add = tu,
+#'   filter_add = TUTESTCD == "TUMIDENT" & VISIT == "SCREENING",
+#'   condition = TUSTRESC == "TARGET",
+#'   false_value = "N",
+#'   missing_value = "N",
+#'   set_values_to = vars(
+#'     PARAMCD = "MDIS",
+#'     PARAM = "Measurable Disease at Baseline"
+#'   )
+#' )
+#' set_admiral_options(set_default = TRUE)
+set_admiral_options <- function(subject_keys,
+                                # future_input,
+                                set_default = FALSE) {
+  if (!missing(subject_keys)) {
+    assert_vars(subject_keys)
+    admiral_env$admiral_options$subject_keys <- subject_keys
+  }
+
+  # if (!missing(future_input)) {
+  #   assert_vars(future_input) nolint
+  #   admiral_env$admiral_options$future_input = future_input nolint
+  # }
+
+  assert_logical_scalar(set_default)
+  if (set_default == TRUE) {
+    admiral_env$admiral_options$subject_keys <- vars(STUDYID, USUBJID)
+    # admiral_env$admiral_options$future_input <- vars(...) nolint
+
+    ######################################################
+    ### To enhance feature and add inputs as necessary ###
+    ######################################################
+
+    # 1. Add additional options such as future_input as shown commented above
+    # 2. Update @params with future_input in set_admiral_options roxygen documentation
+  }
```

Comment:
I would move the setting of default values for options outside this function and directly to where you create the environment. Then you can get rid of the `set_default` parameter.

Reply:
Although restarting the R session would restore the defaults anyway, the intuition behind having `set_default` parameter was to allow a quick & easy way for the user to "go back" within the same session. This came up because I thought I needed to clean up after myself because in the roxygen documentation examples I used `set_admiral_options(subject_keys = vars(STUDYID, USUBJID2))` and it could carry downstream to other examples for other functions that call `subject_keys = get_admiral_option(subject_keys)` . This actually came up in the initial drafting process when I was using a `options()/getOption()` based solution instead of this current environment-based one.